lib/syncthing/syncthing.go

```diff
@@ -224,9 +225,15 @@ func (a *App) startup() error {
 
 	prevParts := strings.Split(prevVersion, "-")
 	curParts := strings.Split(build.Version, "-")
-	if prevParts[0] != curParts[0] {
+	if rel := upgrade.CompareVersions(prevParts[0], curParts[0]); rel != upgrade.Equal {
 		if prevVersion != "" {
 			l.Infoln("Detected upgrade from", prevVersion, "to", build.Version)
+			// Check and repair metadata and sequences once immediately
```

Comment:
Why not just do checkrepair on every upgrade for now?

Reply:
Doing it on all upgrades including RCs feels like a lot of churn (e.g. considering the 11 RCs last month). And we need it to happen now on v1.4.1-rc.2. So maybe do it on stable updates and v1.4.1-rc.2 explicitly could be a middle way. Or am overthinking and for RC users it's fine to do it on every upgrade?

Reply:
Hmm. I didn't consider that we want it to happen between-rc:s this time. However, I wonder if the correct thing here isn't to actually run this on every upgrade, even rc-to-rc. The criteria then would be to split on `+` and just string-equals compare the thing before the plus (if any).

Reply:
I think the full index transfer is a lot heavier than the checks we do, so if anything that would be the part I think we can remove nowadays...